src/m2ee/config.py

```diff
@@ -89,25 +89,20 @@ def _setup_classpath(self):
 
     def _try_load_json(self, jsonfile):
         logger.debug("Loading json configuration from %s" % jsonfile)
-        fd = None
         try:
-            fd = open(jsonfile)
+            with open(jsonfile) as fd:
+                config = json.load(fd)
+            logger.trace("contents read from %s: %s" % (jsonfile, config))
+            return config
         except Exception as e:
             logger.debug("Error reading configuration file %s: %s; "
                          "ignoring..." % (jsonfile, e))
             return {}
-
-        config = None
-        try:
-            config = json.load(fd)
-        except Exception as e:
+        except ValueError as e:
```

Comment:
If you want the valuerror code to ever be reached, it has to go before except Exception, which eats everything.

Reply:
Oops, I forgot to replace that Exception with IOError. Fixed and rebased.